app/code/Magento/InventoryIndexer/Indexer/SelectBuilder.php

```diff
@@ -59,29 +72,53 @@ public function execute($stockId): Select
             return $select;
         }
 
-        $select = $connection
-            ->select()
-            ->from(
-                ['source_item' => $sourceItemTable],
-                [
-                    SourceItemInterface::SKU,
-                    // Quick fix. Proper solution will be provided in https://github.com/magento-engcom/msi/issues/429
-                    SourceItemInterface::QUANTITY => 'SUM(IF(source_item.' . SourceItemInterface::STATUS . ' = '
-                        . SourceItemInterface::STATUS_OUT_OF_STOCK . ', 0, IF(quantity > 0, quantity, 1)))',
-                ]
-            )
-            ->joinLeft(
-                ['stock_source_link' => $sourceStockLinkTable],
-                sprintf(
-                    'source_item.%s = stock_source_link.%s',
-                    SourceItemInterface::SOURCE_CODE,
-                    StockSourceLink::SOURCE_CODE
-                ),
-                []
-            )
+        $select = $connection->select();
+        $quantityExpression = $select->getConnection()->getCheckSql(
```

Comment:
Cast it to string immediately with `(string)`.

Reply:
Fixed.